### PR TITLE
Cover saturate toggle

### PR DESCRIPTION
**Pull request by Blebowski**
Hi,

this is a MR with some left-overs from previous refactor:

- Couple of bug-fixes that creeped in and were un-recognized since they are related to Coverage report printing that is not checked.
- Removed `flag` from `cover_pair_t` -> Redundant now
- Added run-time saturation also for Toggle Items
- Optimizes toggle items in cases when `count-from-undefined` and `count-from-to-z` are enabled.
- Slightly improves coverage report for PSL assertions
- Added Cover item data into the Coverage report (now that it means how many times an item was covered).
- Added a test for saturation during merging. Testing Saturation on increments during run-time is excessively long.

### CODE REVIEW

File: src/rt/cover.c [hunk 1]
```diff
@@ -45,6 +45,13 @@ enum std_ulogic {
    _DC = 0x8
 };
 
+#define INCS_I32(i32_ptr)                          \
+      do {                                         \
+         int32_t inc = *i32_ptr + 1;               \
+         if (likely(inc > *i32_ptr))               \
+            *i32_ptr = inc;                        \
```
Comment (nickg) [inline src/rt/cover.c:52]: This doesn't work with optimisation turned on as signed types have undefined behaviour on overflow unlike unsigned types which wrap. GCC and and Clang remove the saturation check completely: https://godbolt.org/z/hY7xjaK5T

```suggestion
         if (unlikely(__builtin_add_overflow(*i32_ptr, 1, i32_ptr))) \
            *i32_ptr = INT32_MAX;                                    \
```

File: src/cov/cov-data.c [hunk 2]
```diff
@@ -117,6 +118,17 @@ static cover_src_t get_cover_source(cover_item_kind_t kind, object_t *obj)
       }
    }
 
+   psl_node_t p = psl_from_object(obj);
+   if (p != NULL) {
+      printf("PSL NOT NULL!!!\n");
```
[the remark this answers is not in the document]
Comment (Blebowski) [reply src/cov/cov-data.c]: Srry, debugging left-over.

File: src/rt/cover.c (hunk 1 above)
Comment (Blebowski) [reply src/rt/cover.c]: Thanks. I don't have that much experience with the nuances of the C standard, this is very much appreciated.

File: src/rt/cover.c [hunk 3]
```diff
@@ -45,6 +45,13 @@ enum std_ulogic {
    _DC = 0x8
 };
 
+#define INCS_I32(i32_ptr)                                               \
+      do {                                                              \
+         int32_t inc = *i32_ptr + 1;                                    \
```
Comment (nickg) [inline src/rt/cover.c:50]:
```suggestion
```

**Review by nickg — APPROVED** [review]

**Pull request merged**